src/components/Details/DetailedTrafficList.tsx

```diff
@@ -0,0 +1,261 @@
+      percentError = traffic.rates.httpPercentErr ? Number(traffic.rates.httpPercentErr) : 0;
+    } else {
+      percentError = traffic.rates.grpcPercentErr ? Number(traffic.rates.grpcPercentErr) : 0;
+    }
+
+    return (
+      <TableGrid.Col {...minichartColumnSizes}>
+        <svg width="100%" height="1.5em">
+          <rect x="0" y="20%" width="100%" height="60%" fill="green" />
+          <rect x={`${100 - percentError}%`} y="20%" width={`${percentError}%`} height="60%" fill="red" />
+          <rect
+            x={`${100 - REQUESTS_THRESHOLDS.failure}%`}
+            y="0"
+            width="0.4em"
+            height="100%"
+            fill="silver"
+            stroke="white"
+            strokeWidth="0.2em"
+            strokeDasharray="0,0.4em,1.5em,0.4em,1.5em"
+          />
+        </svg>
+      </TableGrid.Col>
+    );
+  };
+
+  private renderWorkloadColumn = (node: TrafficNode) => {
+    let icon = <Icon type="pf" name="unknown" style={{ paddingLeft: '2em' }} />;
+    let name = <>{node.name}</>;
+
+    if (NodeType.WORKLOAD === node.type) {
+      icon = <Icon type="pf" name="bundle" style={{ paddingLeft: '2em' }} />;
+      name = (
+        <Link to={`/namespaces/${encodeURIComponent(node.namespace)}/workloads/${encodeURIComponent(node.name)}`}>
+          {node.name}
+        </Link>
+      );
+    } else if (NodeType.SERVICE === node.type) {
+      icon = <Icon type="pf" name="service" />;
+      name = (
+        <Link to={`/namespaces/${encodeURIComponent(node.namespace)}/services/${encodeURIComponent(node.name)}`}>
+          {node.name}
+        </Link>
+      );
+    } else if (NodeType.APP === node.type) {
+      icon = <Icon type="pf" name="applications" style={{ paddingLeft: '2em' }} />;
+      name = (
+        <Link to={`/namespaces/${encodeURIComponent(node.namespace)}/applications/${encodeURIComponent(node.name)}`}>
+          {node.name}
+        </Link>
+      );
+      if (node.version) {
+        name = (
+          <Link to={`/namespaces/${encodeURIComponent(node.namespace)}/applications/${encodeURIComponent(node.name)}`}>
+            `${node.name} / ${node.version}`
+          </Link>
+        );
+      }
+    }
+
+    return (
+      <TableGrid.Col {...workloadColumnSizes}>
+        {icon} {name}
+      </TableGrid.Col>
+    );
+  };
+
+  private renderTrafficColumn = (traffic: TrafficData) => {
+    if (traffic.protocol === 'tcp') {
+      return <TableGrid.Col {...trafficColumnSizes}>{Number(traffic.rates.tcp).toFixed(2)}</TableGrid.Col>;
+    } else if (traffic.protocol === '') {
+      return <TableGrid.Col {...trafficColumnSizes}>N/A</TableGrid.Col>;
+    } else {
+      let rps: number;
+      let percentError: number;
+
+      if (traffic.protocol === 'http') {
+        rps = Number(traffic.rates.http);
+        percentError = traffic.rates.httpPercentErr ? Number(traffic.rates.httpPercentErr) : 0;
+      } else {
+        rps = Number(traffic.rates.grpc);
```

Comment:
So, typescript here doesn't show an error due to potentially undefined `traffic.rates`, which is correct, it's able to infer type based on previous conditional branches and matches on `traffic.protocol` ... Nothing to add, I just wanted to say loud how awesome is typescript! :-)